src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -1501,10 +1501,12 @@ void BytecodeInterpreter::run(interpreterState istate) {
 #define ARRAY_INTRO(arrayOff)                                                  \
       arrayOop arrObj = (arrayOop)STACK_OBJECT(arrayOff);                      \
       jint     index  = STACK_INT(arrayOff + 1);                               \
-      char message[jintAsStringSize];                                          \
+      char message[2 * jintAsStringSize + 35];                                 \
```

Comment:
What's this magic `35` here? `strlen("Index A out of bounds for length B") + 1 == 35`. Isn't this counting the integers twice?

Consider adding a comment.

Reply:
Seems OK to overestimate a bit. That's what `Runtime1::throw_range_check_exception` does in C1, so I have not challenged it:
https://github.com/openjdk/jdk/blob/2f06893a29fba3b40cc1cf03095b75b217d1bb93/src/hotspot/share/c1/c1_Runtime1.cpp#L668-L671

It should count the integers twice, because there are two integers in the string.

Reply:
Well, fine, I'll count them exactly and `STATIC_ASSERT` it ;)

Reply:
Done in new revision.